t/03-testapi.t

```diff
@@ -441,23 +441,50 @@ sub script_output_test {
 
     $mock_testapi->mock(wait_serial => sub { return "XXXfoo\nSCRIPT_FINISHEDXXX-0-" });
     is(script_output('echo foo'), 'foo', 'sucessfull retrieves output of script');
+    my @array = script_output('echo foo');
+    is(@array,    1,     'No auto array context 1');
+    is($array[0], 'foo', 'No auto array context 2');
+    my $retval;
+    my $out = script_output('echo foo', exit_code => \$retval);
+    is($out,    'foo', 'sucessfull retrieves output of script (wantarray)');
+    is($retval, 0,     'sucessfull retrieves output of script (wantarray)');
 
     $mock_testapi->mock(wait_serial => sub { return 'SCRIPT_FINISHEDXXX-0-' });
     is(script_output('foo'), '', 'calling script_output does not fail if script returns with success');
+    $retval = undef;
+    $out    = script_output('foo', exit_code => \$retval);
+    is($out,    '', 'calling script_output does not fail if script returns with success (wantarray)');
+    is($retval, 0,  'calling script_output does not fail if script returns with success (wantarray)');
 
     $mock_testapi->mock(wait_serial => sub { return "This is simulated output on the serial device\nXXXfoo\nSCRIPT_FINISHEDXXX-0-\nand more here" });
     is(script_output('echo foo'), 'foo', 'script_output return only the actual output of the script');
+    $retval = undef;
+    $out    = script_output('foo', exit_code => \$retval);
```

Comment:
Oh, this is really C like approach :). I'm not sure about it, I'd prefer previous version which returns array or scalar, even it's "considered bad practice" :).